CRM/Utils/Recent.php

```diff
@@ -50,9 +50,19 @@ class CRM_Utils_Recent {
   static private $_recent = NULL;
 
   /**
+   * Maximum stack size
+   * @var int
+   */
+  static private $_maxItems = 20;
+
+  /**
    * Initialize this class and set the static variables.
    */
   public static function initialize() {
+    $maxItemsSetting = Civi::settings()->get('recentItemsMaxCount');
+    if (isset($maxItemsSetting) && $maxItemsSetting > 0 && $maxItemsSetting < 100) {
```

Comment:
So if you don't allow anyone to insert value not in 1-100 range , then this checking part is not necessary